src/domain/service/noteSettings.ts

```diff
@@ -31,6 +33,24 @@ export default class NoteSettingsService {
   constructor(noteSettingsRepository: NoteSettingsRepository, teamRepository: TeamRepository, private readonly shared: SharedDomainMethods) {
     this.noteSettingsRepository = noteSettingsRepository;
     this.teamRepository = teamRepository;
+
+    /**
+     * Listen to the note related events
+     */
+    EventBus.getInstance().addEventListener(NOTE_ADDED_EVENT_NAME, async (event) => {
+      const { noteId, userId } = (event as CustomEvent<{ noteId: number; userId: number }>).detail;
```

Comment:
can we get rid of this casting? Maybe we can declare EventMap. Example:

https://github.com/editor-js/document-model/blob/main/packages/model/src/EventBus/types/EventMap.ts
https://github.com/editor-js/document-model/blob/main/packages/model/src/EventBus/types/EventTarget.d.ts